.travis.yml

```diff
@@ -2,23 +2,18 @@ language: python
 
 matrix:
   include:
-    - python: 3.4
     - python: 3.5
     - python: 3.6
     - python: 3.7
-      dist: xenial
-      sudo: true
-    - python: 3.7-dev
-      dist: xenial
-      sudo: true
     - python: 3.8-dev
       dist: xenial
       sudo: true
 
 install:
-  - pip install --upgrade setuptools wheel
-  - python setup.py install
```

Comment:
I have a mild concern: This project now has multiple independent build systems. Using `setup.py` here was a deliberate choice to ensure the CI checks used the exact same build process as the release. With this change it's now possible to PR a broken change to `setup.py` with all CI checks passing.

So not something to need to change now, but I'd like to see this project move towards a single build/test/release system. I don't care which one, but let's be opinionated and stick with it.

Reply:
This is using the `setup.py` file to install (the `pip install .` command). See https://pip.pypa.io/en/latest/reference/pip_install/#local-project-installs for more details.

Reply:
Ah cool thanks. I didn't know that.

Reply:
Those both sound sensible to me, though I don't really have enough experience of production Python to know for sure.

Reply:
1) Next time we need to do a release I can help with using the Makefile to build both distribution types and uploading them to pypi with twine

2): I'll leave off moving the package into `/src` until after @pfairfax1  PR is merged.  This change will only impact how we test the project.  The user experience and developer experience will both be the same.